msal-python-conceptual/advanced/macos-broker.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Using MSAL Python with Authentication Brokers on macOS
+description: "Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as token binding, protecting any issued tokens from exfiltration and misuse."
+author: localden
+manager: CelesteDG
+
+ms.service: msal
+ms.subservice: msal-python
+ms.topic: conceptual
+ms.date: 02/07/2024
+ms.author: ddelimarsky
+ms.reviewer: shermanouko, rayluo
+---
+
+# Using MSAL Python with Authentication Brokers on macOS
+
+>[!NOTE]
+>macOS authentication broker support is introduced with `msal` version 1.31.0.
+
+Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as **token binding**, protecting any issued tokens from exfiltration and misuse.
+
+Authentication brokers are **not** pre-installed on macOS but are applications developed by Microsoft, such as [Company Portal](/mem/intune/apps/apps-company-portal-macos). These applications are usually installed when a macOS computer is enrolled in a company's device fleet via an endpoint management solution like [Microsoft Intune](/mem/intune/fundamentals/what-is-intune). To learn more about Apple device set up with the Microsoft Identity Platform, refer to [Microsoft Enterprise SSO plug-in for Apple devices](/entra/identity-platform/apple-sso-plugin).
+
+Typically, on macOS your [public client](/entra/identity-platform/msal-client-applications) Python applications would [acquire tokens](../getting-started/acquiring-tokens.md) via the system browser. To use authentication brokers installed on a macOS system instead, you will need to pass an additional argument in the `PublicClientApplication` constructor - `enable_broker_on_mac`:
+
+```python
+from msal import PublicClientApplication
+ 
+app = PublicClientApplication(
+    "CLIENT_ID",
+    authority="https://login.microsoftonline.com/common",
+    enable_broker_on_mac =True)
+```
+
+>[!IMPORTANT]
+>If you are writing a cross-platform application, you will also need to use `enable_broker_on_windows`, as outlined in the [Using MSAL Python with Web Account Manager](wam.md) article.
+
+>[!NOTE]
+>Integration with the macOS authentication brokers is currently only available for **scripts and non-GUI** applications written in Python.
+
+In addition to the constructor change, your application needs to support broker-specific redirect URIs. For _unsigned_ applications, the URI is:
+
+```text
+msauth.com.msauth.unsignedapp://auth 
+```
+
+For signed applications, the redirect URI should be:
+
+```text
+msauth.BUNDLE_ID://auth
+```
```

Comment:
@fengga mentioned it offline that a Python script might not be able to have bundle id at all. If so, that this "for signed applications" paragraph is not applicable to be put in this "MSAL Python Conceptual" document.

Reply:
@fengga @rayluo can it _ever_ be signed? What about GUI apps?